### ISSUE TITLE
build_and_push_image not working with custom "gcloud-service-key" var

### ISSUE THREAD

**Issue by philicious** [CircleCI-Public/gcp-gcr-orb#12]
### Orb version

0.0.5 (0.0.7 still affected according to orb code)

### What happened

```
  build_and_push_image:
    jobs:
      - gcp-gcr/build_and_push_image:
          gcloud-service-key: GCLOUD_SERVICE_KEY_CUSTOM
```
this does not work as the `gcp-cli/initialize` step still uses the default env-var name
>  Store service account
> echo $GCLOUD_SERVICE_KEY > ${HOME}/gcloud-service-key.json
### Expected behavior

the env-var name from parameter is used.

parameter needs to be passed on here https://github.com/CircleCI-Public/gcp-gcr-orb/blob/fdfeca868cfd43d200fc64bb3e12f3704c085042/src/%40orb.yml#L162 and here https://github.com/CircleCI-Public/gcp-gcr-orb/blob/fdfeca868cfd43d200fc64bb3e12f3704c085042/src/%40orb.yml#L56

this is partially fixed in the stale PRs #5 and #3

**Comment by iynere** [comment]
can you confirm that this is still an issue in `circleci/gcp-gcr@0.0.7` ? @philicious

**Comment by philicious** [comment]
@iynere (huh, wondering why I didnt use 0.0.7. Must have copied from some example in Circle docs)
no matter what, I just checked master HEAD and the lines that need to be fixed are still the same. So latest version 0.0.7 is also affected.

**Comment by iynere** [comment]
ahhh, so if i'm undrestanding correctly, the issue is with the `gcp-cli-orb` not this one, yes ? @philicious 

if so, can you file this issue against that repo ?

http://github.com/circleci-public/gcp-cli-orb

**Comment by philicious** [comment]
@iynere no its in this orb here. when invoking `gcp-gcr/build_and_push_image`, it doesnt forward a custom set `gcloud-service-key` parameter to `gcp-gcr/gcp-auth` which then also misses to forward it to `gcp-cli/initialize` . so the latter then runs with default value `$GCLOUD_SERVICE_KEY` 

as a result, you cannot use multiple `build_and_push_image` workflows with different keys for pushing to different registries that require different service-keys

I could make a PR fixing the lines, but I have no clue about your testing-setup. However I mentioned the lines where you need to insert the parameter-forwarding in my first post

**Comment by iynere** [comment]
ahhh i see, the problem is that `gcr-auth` _takes_ a `gcloud-service-key` parameter but then doesn't do anything with it, so of course no other commands / jobs will either

**Comment by iynere** [comment]
should be patched in `circleci/gcp-gcr@0.1.0` https://circleci.com/orbs/registry/orb/circleci/gcp-gcr @philicious

**Comment by philicious** [comment]
@iynere thanks for the work. however during this destructure/refactor, I think sth broke:

I now get 
>  Error calling workflow: 'build_and_push_image'
>  Cannot find a definition for job named gcp-gcr/build_and_push_image
and the entire job is broken.

also here, we still miss to pass on the service-key parameter
https://github.com/CircleCI-Public/gcp-gcr-orb/blob/d0929c38fe568af91492edf424d99326a8438dba/src/jobs/build-and-push-image.yml#L45

for referenfe, here is my config.yml
```yaml
version: 2.1
orbs:
  gcp-gcr: circleci/gcp-gcr@0.1.0

workflows:
  build_and_push_image:
    jobs:
      - gcp-gcr/build_and_push_image:
          registry-url: eu.gcr.io
          image: gcloud-build
          tag: SDK-224.0.0
          filters:
            branches:
              only:
                - master
  build_and_push_image_foobar:
    jobs:
      - gcp-gcr/build_and_push_image:
          gcloud-service-key: GCLOUD_SERVICE_KEY_FOOBAR
          google-project-id: GOOGLE_PROJECT_ID_FOOBAR
          registry-url: eu.gcr.io
          image: gcloud-build
          tag: SDK-224.0.0
          filters:
            branches:
              only:
                - master
```
(the first workflow uses default vars set as project-wide env-vars, while the second one uses a different service-key and gcp-project)

**Comment by iynere** [comment]
I changedthe name to `build-and-push-image` for consistency with our other orbs @philicious

**Comment by iynere** [comment]
ahhh thank you,will fix that issue on line 45

**Comment by iynere** [comment]
fixed in `circleci/gcp-gcr@0.2.0` @philicious

**Comment by philicious** [comment]
@iynere thx. I can report `circleci/gcp-gcr@0.2.0` fixes this issue. 👍
